### PR TITLE
Add JSON filter to upload, replace, upload_picture and upload_texttrack methods.

### PR DESCRIPTION
**Pull request by etienned**
These changes will help avoid hitting the rate limit. It's possible to specified other fields wanted in `upload_picture` and `upload_textrack` methods with the `fields` argument.

I also made a small change that make vimeo.py compatible with python 2.6 (unfortunately I currently need this, but soon that will be move to 2.7 and 3.6 :-( ).

### CODE REVIEW

File: vimeo/upload.py [hunk 1]
```diff
@@ -21,19 +22,23 @@ class UploadVideoMixin(object):
...
     def upload(self, filename, upgrade_to_1080=False):
         """Upload the named file to Vimeo."""
-        ticket = self.post(self.UPLOAD_ENDPOINT,
-                data={'type': 'streaming',
-                      'upgrade_to_1080': 'true' if upgrade_to_1080 else 'false'})
+        ticket = self.post(
+            self.UPLOAD_ENDPOINT,
+            data={'type': 'streaming',
+                  'upgrade_to_1080': 'true' if upgrade_to_1080 else 'false'},
+            params={'fields': 'upload_link,complete_uri'})
```
Comment (erunion) [inline vimeo/upload.py]: It'd be preferable if these weren't passed in as defaults. The default library behavior always should be to pull everything back from a resource, and then if you want to slim down your payload, you can supply a list of the only fields you want.

Maybe an just add a new `fields` parameter to these `upload` and `upload_picture` methods, similar to how you did with `upload_texttrack` (just without hardcoding a set of base fields).

Comment (erunion) [inline vimeo/upload.py]: Talked it over with the rest of the team, and it's okay. 👍

**Comment by erunion** [comment]
Tagged this to 0.3.10. Thanks!

File: vimeo/upload.py (hunk 1 above)
Comment (etienned) [inline vimeo/upload.py]: Great!

**Pull request merged**